qos/solana/observe.go

```diff
@@ -0,0 +1,84 @@
+// solana package provides the support required for interacting
+// with the Solana blockchain through the gateway.
+package solana
+
+import (
+	"errors"
+
+	"github.com/buildwithgrove/path/message"
+	"github.com/buildwithgrove/path/relayer"
+)
+
+// observationSet provides all the functionality required
+// by the message package's ObservationSet to handle the sharing
+// of QoS data between PATH instances, and the updating of local
+// PATH instance's QoS data on Solana endpoints.
+var _ message.ObservationSet = observationSet{}
+
+// observation captures the result of processing an endpoint's
+// response to a service request.
+// It provides details needed to establish the validity of
+// an endpoint.
+type observation interface {
+	// Apply updates the endpoint based on the observation's contents.
```

Comment:
This is great, love the simplicity of providing the `Apply` method for updating endpoint results.